libc/libc_test.go

```diff
@@ -0,0 +1,555 @@
+				0xc1, 0xe0, 0x04, 0x64, 0x48, 0x03, 0x04, 0x25,
+				0x10, 0x00, 0x00, 0x00, 0x48, 0x8b, 0x50, 0x08,
+				0x48, 0x85, 0xd2, 0x74, 0x53, 0x89, 0xff, 0x48,
+				0x8d, 0x0d, 0x92, 0x7b, 0x16, 0x00, 0x48, 0x8b,
+				0x30, 0x48, 0xc1, 0xe7, 0x04, 0x48, 0x39, 0x34,
+				0x39, 0x75, 0x35, 0x48, 0x89, 0xd0, 0xc3,
+			},
+			info: TSDInfo{
+				Offset:     0x310 + 8,
+				Multiplier: 0x10,
+			},
+		},
+		"glibc 2.38 / Fedora 39 / arm64": {
+			machine: elf.EM_AARCH64,
+			code: []byte{
+				0x3f, 0x23, 0x03, 0xd5, // paciasp
+				0xfd, 0x7b, 0xbf, 0xa9, // stp     x29, x30, [sp, #-16]!
+				0xe1, 0x03, 0x00, 0x2a, // mov     w1, w0
+				0xfd, 0x03, 0x00, 0x91, // mov     x29, sp
+				0x1f, 0x7c, 0x00, 0x71, // cmp     w0, #0x1f
+				// b.hi    91d98 <__pthread_getspecific@GLIBC_2.17+0x58>  // b.pmore
+				0x28, 0x02, 0x00, 0x54,
+				// mov     x0, #0xfffffffffffff9d0         // #-1584
+				0xe0, 0xc5, 0x80, 0x92,
+				0x42, 0xd0, 0x3b, 0xd5, // mrs     x2, tpidr_el0
+				0x00, 0x50, 0x21, 0x8b, // add     x0, x0, w1, uxtw #4
+				0x42, 0x00, 0x00, 0x8b, // add     x2, x2, x0
+				0x40, 0x04, 0x40, 0xf9, // ldr     x0, [x2, #8]
+				// cbz     x0, 91dcc <__pthread_getspecific@GLIBC_2.17+0x8c>
+				0x00, 0x03, 0x00, 0xb4,
+				0x21, 0x7c, 0x7c, 0xd3, // ubfiz   x1, x1, #4, #32
+				0x83, 0x09, 0x00, 0xb0, // adrp    x3, 1c2000 <initial+0x198>
+				0x63, 0x40, 0x17, 0x91, // add     x3, x3, #0x5d0
+				0x44, 0x00, 0x40, 0xf9, // ldr     x4, [x2]
+				0x61, 0x68, 0x61, 0xf8, // ldr     x1, [x3, x1]
+				0x3f, 0x00, 0x04, 0xeb, // cmp     x1, x4
+				// b.ne    91dc8 <__pthread_getspecific@GLIBC_2.17+0x88>  // b.any
+				0x01, 0x02, 0x00, 0x54,
+				0xfd, 0x7b, 0xc1, 0xa8, // ldp     x29, x30, [sp], #16
+				0xbf, 0x23, 0x03, 0xd5, // autiasp
+				0xc0, 0x03, 0x5f, 0xd6, // ret
+				// code skipped handling keys >0x1f
+			},
+			info: TSDInfo{
+				Offset:     -1584 + 8,
+				Multiplier: 16,
+			},
+		},
+		"booking coredump glibc": {
+			machine: elf.EM_X86_64,
+			code: []byte{
+				0x83, 0xff, 0x1f, 0x77, 0x49, 0x89, 0xf8, 0x48, 0x83, 0xc0, 0x30, 0x48,
+				0xc1, 0xe0, 0x04, 0x64, 0x48, 0x8b, 0x14, 0x25, 0x10, 0x00, 0x00, 0x00,
+				0x48, 0x8d, 0x54, 0x02, 0x10, 0x48, 0x8b, 0x42, 0x08, 0x48, 0x85, 0xc0,
+				0x74, 0x1a, 0x89, 0xff, 0x48, 0x8d, 0x0d, 0x61, 0xaa, 0x20, 0x00, 0x48,
+				0xc1, 0xe7, 0x04, 0x48, 0x8b, 0x34, 0x39, 0x48, 0x39, 0x32, 0x75, 0x07,
+				0xf3, 0xc3,
+			},
+			info: TSDInfo{
+				Offset:     0x310 + 8,
+				Multiplier: 0x10,
+			},
+		},
+	}
+
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var info TSDInfo
+			var err error
+			switch test.machine {
+			case elf.EM_X86_64:
+				info, err = extractTSDInfoX86(test.code)
+			case elf.EM_AARCH64:
+				info, err = extractTSDInfoARM(test.code)
+			}
+			if assert.NoError(t, err) {
+				assert.Equal(t, test.info, info, "Wrong TSD info extraction")
+			}
+		})
+	}
+}
+
+func TestExtractDTVOffset(t *testing.T) {
```

Comment:
This is the actual new code containing the tests for disassembly of `__tls_get_addr` on different combinations of glibc, musl, x86_64, and aarch64.

Reply:
Thanks, my bad on missing this. Seems indeed that Github hid it by default.